lib/ResolverFactory.js

```diff
@@ -29,13 +29,19 @@ const EMPTY_RESOLVE_OPTIONS = {};
 module.exports = class ResolverFactory {
```

Comment:
revert this changes in this file, in favor of having `dependencyType` as part of `resolveOptions`